Border color for stacked windows in stack layout

Jack8472 opened the pull request:
This adds a new feature in **stack layout**. Now it is possible to change border color if the windows are in 'split' or 'stacked' state. This feature is present in column layout and is very useful to quickly see if there are windows "hiding" below. The feature will be as useful in stack layout.

elParaguayo commented:
Can you post a screenshot to explain what this does?

Also, when adding new options, we should try not to change the default behaviour/appearance i.e. users who upgrade will not expect to see something different.

Jack8472 commented:
I made three screenshots with extra wide borders which hopefully convey the behavior of the functionality. This behavior is already present in the column layout so maybe it is not a big change in behavior.

![Screenshot-1-windows-are-split png](https://user-images.githubusercontent.com/3611452/194701006-d071a739-4c52-44c1-933f-610a8d936a21.png)
![Screenshot-2-windows-are-stacked-focused](https://user-images.githubusercontent.com/3611452/194701013-9e1eaa10-aaf3-476e-a318-c68770e4d004.png)
![Screenshot-3-windows-are-stacked-unfocused png](https://user-images.githubusercontent.com/3611452/194701017-8ff4f893-a454-479b-a255-9704ce89d9a2.png)

Jack8472 commented:
Similar functionality in columns layout I was refering to above is in [columns.py](https://github.com/qtile/qtile/blob/master/libqtile/layout/columns.py):

Lines 121-130:

```
        (
            "border_focus_stack",
            "#881111",
            "Border colour(s) for the focused window in stacked columns.",
        ),
        (
            "border_normal_stack",
            "#220000",
            "Border colour(s) for un-focused windows in stacked columns.",
        ),
```

Lines 250-253:

```
        if client.has_focus:
            color = self.border_focus if col.split else self.border_focus_stack
        else:
            color = self.border_normal if col.split else self.border_normal_stack
```

m-col commented on libqtile/layout/stack.py:
```diff
@@ -55,6 +55,10 @@ class Stack(Layout):
     defaults = [
         ("border_focus", "#0000ff", "Border colour(s) for the focused window."),
         ("border_normal", "#000000", "Border colour(s) for un-focused windows."),
+        ("border_focus_stack", "#881111",
+         "Border colour(s) for the focused stacked window."),
+        ("border_normal_stack", "#220000",
+         "Border colour(s) for un-focused stacked windows."),
```
Could you make the default values `None`? That way the feature is opt-in and users won't have things suddenly change when they update.

Then below when the colors are accessed it could test for `None`s.

Jack8472 replied:
I tried to do what you suggested. Please let me know if I was successful.

m-col reviewed (changes requested):
Neat! I've made a small comment inline but otherwise it's a great feature, thanks for the submission!

m-col reviewed (commented):
Thanks!

I'm also wondering about the docstrings for the options: "Border colour(s) for the focused stacked window." It might be worth making it clear that if not provided, it will fall back to the `border_focus`?

Also - if you add an entry to the changelog, users will learn about this option and can use it :)

Jack8472 commented on libqtile/layout/stack.py:
```diff
@@ -206,9 +210,9 @@ def configure(self, client, screen_rect):
 
         if client.has_focus:
-            px = self.border_focus
+            px = self.border_focus if s.split else self.border_focus_stack
```
Thank you. Did my best to follow the suggestions.

Jack8472 commented:
Had to do another committ to fix a bug (colors were reversed). Now the thing works as intended.

m-col reviewed (approved):
Perfect :) I've triggered the CI and will enable auto-merge so that if all the tests pass it will get merged.

Thanks for the PR!

m-col commented:
Not very clear error message:
```
  libqtile/layout/stack.py:60:10: BLK100 Black would make changes.
```
I suspect it doesn't like the way the multi-line doc strings are formatted in some way. Might be worth running `black` on the libqtile folder and let it figure it out?

Jack8472 commented:
> I suspect it doesn't like the way the multi-line doc strings are formatted in some way. Might be worth running `black` on the libqtile folder and let it figure it out?

Yes. Black found bad formatting on my strings. Sorry for that. Committed a correction.

m-col commented:
Fantastic, thanks again!

Jack8472 commented:
> Fantastic, thanks again!

Thank YOU very much!
